### PR TITLE
linter-tutorial

### PR DESCRIPTION
**Pull request by kghamilton89**
@raipc This article is meant to explain how to pull our Travis CI config and apply it to your local repository, for example a company which wants to use our lint rules etc. Almost everything worked fine as I completed the process but I have two questions:

1. `linkcheck` feature is not working in my repo, I see there's a [if/else](https://github.com/axibase/atsd-use-cases/blob/c140dd99c943b6661fb713a39ddf4a0ab7db2441/tutorials/linter-tutorial/resources/.travis-functions.sh#L34) statement which looks like it controls the behavior, but I don't know how to modify it so that the check is active. Note that the line I modified (37) was unchanged when I was testing.
2. When I set up the checker on my local repo, I automatically began to receive email and rocket.chat notifications. It looks like this is done via webhook. So my question is this: Does the user have to set anything up themselves or is the email read from the user account and automatically passed to ATSD?

Thank you.

PS: Any suggestions you have about how to better express any part of the tutorial are also welcome.

### CODE REVIEW

**Comment by raipc** [comment]
1. I could not reproduce the problem.
https://github.com/kghamilton89/demo-repo/pull/24

2. Check `notifications > webhooks` in .travis.yml. Here you set the ATSD message webhook URL which later will be processed by `travis-ci-build-status` rule and send notifications to the committer based on email specified in webhook body.

File: tutorials/linter-tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+The `.travis.yml` file references the `.travis-functions.sh` file which controls certain components of the tool such as the repository language and custom dictionary.
+
+```sh
+spellchecker --language=en-US
+```
+
+Download the [`.travis-functions.sh`](./resources/.travis-functions.sh) file from the attached `/resources` directory and upload it to your monitored repositories.
+
+After uploading the `.travis.yml` and `travis-functions.sh` files to your local repository, the tool is operational. New builds are triggered upon the opening of a new pull request, or any push made directly to a repository such as those performed by a repository administrator.
```
Comment (raipc) [inline tutorials/linter-tutorial/README.md]: .travis-functions.sh

File: tutorials/linter-tutorial/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+## Configuring Custom Rules
+
+The attached configuration contains a number of [custom rules](https://github.com/axibase/docs-util#custom-rules) which reflect the guidelines outlined in the [Axibase Technical Writing Style Guide](../workshop/technical-writing.md). Modify existing custom rules or create your own according to the instructions in the [How To Add Custom `markdownlint` Rules Tutorial](https://github.com/axibase/docs-util/blob/master/docs/how-to-add-custom-linter-rule.md#how-to-add-custom-markdownlint-rule).
+
+### Custom Dictionaries
+
+By default, the Travis CI tool is configured to use words included in the `.dictionary-*` files. These documents are divided into two categories: [**Names**](https://github.com/axibase/atsd/blob/master/.dictionary-names) and [**Other**](https://github.com/axibase/atsd/blob/master/.dictionary-other). Words in `.dictionary-names` are **case sensitive** and must be expressed with letters capitalized exactly as they are in the dictionary. Words in `.dictionary-other` can be expressed in upper or lower case as the situation requires and can optionally include syntax which allows plural forms to be used.
```
Comment (raipc) [inline tutorials/linter-tutorial/README.md]: I think it's worth to mention that these dictionaries are generated by the [python script](https://raw.githubusercontent.com/axibase/docs-util/master/python-scripts/dictionaries_generator.py) and contains some abbreviations widely used in our docs repositories. The words from the dictionaries are converted to regular expressions recognized by spell checking tools.

File: tutorials/linter-tutorial/README.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+```JSON
+{
+    "ignorePatterns":[
+        {"pattern": "https://example.org"},
+    ]
+}
+```
+
+Update the list to include those URL patterns which are to be ignored by the tool during the `linkcheck` phrase of the build process.
```
Comment (raipc) [inline tutorials/linter-tutorial/README.md]: `linkcheck` phase

File: tutorials/linter-tutorial/README.md (hunk 2 above)
Comment (raipc) [reply tutorials/linter-tutorial/README.md]: Also, in Axibase repositories dictionaries are concatenated with base dictionary located in `axibase/atsd` repository (hardcoded in abovementioned script).

File: tutorials/linter-tutorial/README.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+On line 37 of `.travis-functions.sh`, define the literal URL link to the file whose contents include the enumeration of links to be ignored by Travis CI.
```
Comment (raipc) [inline tutorials/linter-tutorial/README.md]: On line 37 the location of default `.linkcheck-config.json` is given. It is used if this file is not found locally.

File: tutorials/linter-tutorial/README.md (hunk 1 above)
Comment (kghamilton89) [reply tutorials/linter-tutorial/README.md]: updated.

File: tutorials/linter-tutorial/README.md (hunk 2 above)
Comment (kghamilton89) [reply tutorials/linter-tutorial/README.md]: mentioned. good tip.

File: tutorials/linter-tutorial/README.md (hunk 3 above)
Comment (kghamilton89) [reply tutorials/linter-tutorial/README.md]: typo repaired

File: tutorials/linter-tutorial/README.md (hunk 4 above)
Comment (kghamilton89) [reply tutorials/linter-tutorial/README.md]: Huh. Did not know that. Good tip thank you.

**Pull request merged**